### ISSUE TITLE
WCAG 2.1 compliancy: Disable keyboard controls for embedded player

### ISSUE THREAD

**Issue by bskibinski** [vimeo/player.js#732]
### Expected Behavior
A parameter for the embedded Vimeo player to disable keyboard controls similar to Youtube's `disablekb`
https://developers.google.com/youtube/player_parameters

This is needed to be WCAG 2.1 compliant (accessibility).
https://www.w3.org/WAI/WCAG21/Understanding/character-key-shortcuts

### Actual Behavior
No such parameter exists for Vimeo:
https://vimeo.zendesk.com/hc/en-us/articles/360001494447-Using-Player-Parameters
https://developer.vimeo.com/player/sdk/embed

### Steps to Reproduce
Visit https://vimeo.zendesk.com/hc/en-us/articles/360001494447-Using-Player-Parameters
And you will see there exists no parameter to disable keyboard controls for the embedded player.

This prevents using the player on websites that require WCAG2.1 compliancy level A (the lowest rating).

Even better would be to make the controls disabled by default, and add the parameter to enable keyboard controls.

**Comment by rianrietveld** [comment]
Thank you @bskibinski, I agree with this 100%.

When embedding a Vimeo video on a webpage, the video has shortcuts, for e.g. to like, set fullscreen etc.
These are one character shortcuts. These shortcuts can not be disabled.

It is important this shortcuts can be disables, because they hinder the shortcuts of assistive technology like a screen reader.

This is also a violation of  the Web Content Accessibility Guidelines (WCAG), version 2.1, level A: [Success Criterion 2.1.4: Character Key Shortcuts](https://www.w3.org/WAI/WCAG21/quickref/#character-key-shortcuts).

This Success Criterion states:
If a keyboard shortcut is implemented in content using only letter (including upper- and lower-case letters), punctuation, number, or symbol characters, then at least one of the following is true:
- Turn off: A mechanism is available to turn the shortcut off;
- Remap: A mechanism is available to remap the shortcut to use one or more non-printable keyboard characters (e.g. Ctrl, Alt, etc);
- Active only on focus: The keyboard shortcut for a user interface component is only active when that component has focus.

All of these options are not available in the Vimeo settings at the moment.

Therefore Vimeo videos can not be used for websites that need to comply to WCAG 2.1. Vimeo can not be used  for government and public service websites in the EU and the UK at the moment. And many more in the future.

Please add a way to disable the shortcuts, 
Like @bskibinski said: just like the option YouTube offers with the parameter `disablekb=1` or by making the controls disabled by default, and add the parameter to enable keyboard controls.

More information can be found on the W3C website:
[Understanding Success Criterion 2.1.4: Character Key Shortcuts](https://www.w3.org/WAI/WCAG21/Understanding/character-key-shortcuts.html)

Related https://github.com/vimeo/player.js/discussions/673

**Comment by anneke** [comment]
+1! Thank you @bskibinski @rianrietveld

**Comment by jeroensmeets** [comment]
Also a +1 from me, great suggestion and explained clearly.

**Comment by Aircl0wn** [comment]
+1 for adding this.

**Comment by NMHeckel** [comment]
One more in favor of this! Please make Vimeo WCAG 2.1 conformant and usable for people who navigate with screen readers!

**Comment by sinabahram** [comment]
+1 to this. It would be good for Vimeo to continue the existing efforts around accessibility, of which this issue is just one necessary piece. Thanks to those contributing to such important and meaningful work.

**Comment by NicolaSaunders** [comment]
+1 to this

**Comment by wpalchemist** [comment]
+1

**Comment by rkrishnan8594** [comment]
Hey all - I want to thank everyone for raising awareness about this gap in the Vimeo API and its impact on WCAG compliance. It's important to us that the player is at the forefront of accessibility standards and that our libraries enable developers to build accessible experiences.

To that end, we have this issue on our radar and a solution involving adding a new parameter is queued up for development work within the next 1-2 weeks. I'll keep this thread posted as we make progress and have more updates to share.

**Comment by rianrietveld** [comment]
@rkrishnan8594 thank you so much!

**Comment by hidde** [comment]
Fantastic, @rkrishnan8594!

**Comment by MatteoPieroni** [comment]
💯 agree, especially for a video-focused company like Vimeo this is super important

**Comment by mraccess77** [comment]
Seems like you only need a feature to prevent the single key shortcuts from working out the player (when the player is not focused) - they could continue to work inside the player.

**Comment by KoenWouters** [comment]
Thank you @rkrishnan8594 🥳 . Hope to see this update soon 😄

**Comment by julezrulez** [comment]
> Seems like you only need a feature to prevent the single key shortcuts from working out the player (when the player is not focused) - they could continue to work inside the player.

That's what I thought too the first time I read success criterium 2.1.4. But, it says actually "user interface component" which is for instance one of the controls. The whole player is not seen as a user interface component. So for this criterium the "F" for full screen might only work if the focus is on the full-screen button. Nowhere else.

**Comment by ShadowBB** [comment]
+ 1

Also agree with Julez and his interpretation of "user interface component" in WCAG. That is how we interpret it during WCAG audits.

**Comment by bskibinski** [comment]
Thanks @rianrietveld, @julezrulez and @ShadowBB and everyone else for the excellent additions. This issue blew up overnight.

And @rkrishnan8594 great to hear this is getting picked up this fast, good luck!

**Comment by brandonhrowe** [comment]
Thank you everyone for raising this issue and providing feedback, which has been extremely helpful in finding a solution. We are glad to say that the issue is now resolved with version 2.16.0 of player.js, which can now take in an optional `keyboard` option. If `keyboard` is set to `false` or `0`, it will disable keyboard input on the player. Thank you all again for providing input on this functionality!

**Comment by rianrietveld** [comment]
@brandonhrowe thank you so much!

**Comment by bskibinski** [comment]
Super awesome! Some municipalities of the Netherlands will be grateful for this ;) thx for the quick fix!

**Comment by bskibinski** [comment]
@brandonhrowe I think this page still has to be updated with the new paramater:
https://vimeo.zendesk.com/hc/en-us/articles/360001494447-Using-Player-Parameters

**Comment by rkrishnan8594** [comment]
Thanks for the note @bskibinski, we will work on updating that page shortly.

**Comment by rkrishnan8594** [comment]
That page should now be updated, @bskibinski.

**Comment by MakoMori-37** [comment]
many thank!
